kms/asymmetric/samples_test.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/cloudkms/v1"
+	"testing"
+)
+
+type TestVariables struct {
+	client         *cloudkms.Service
+	ctx            context.Context
+	message        string
+	rsaDecryptPath string
+	rsaSignPath    string
+	ecSignPath     string
+}
+
+func setup(t *testing.T) TestVariables {
+	tc := testutil.SystemTest(t)
+	projectId := tc.ProjectID
+	parent := "projects/" + projectId + "/locations/global"
+	keyRing := "kms-asymmetric-samples"
+
+	rsaDecrypt := parent + "/keyRings/" + keyRing + "/cryptoKeys/rsa-decrypt/cryptoKeyVersions/1"
+	rsaSign := parent + "/keyRings/" + keyRing + "/cryptoKeys/rsa-sign/cryptoKeyVersions/1"
+	ecSign := parent + "/keyRings/" + keyRing + "/cryptoKeys/ec-sign/cryptoKeyVersions/1"
+
+	message := "test message 123"
+
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, cloudkms.CloudPlatformScope)
+	if err != nil {
+		t.Fatal(err)
+	}
+	kmsClient, err := cloudkms.New(client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//create cryptokeys
+	kmsClient.Projects.Locations.KeyRings.Create(parent, &cloudkms.KeyRing{}).KeyRingId(keyRing).Do()
+	kmsClient.Projects.Locations.KeyRings.CryptoKeys.Create(
+		parent+"/keyRings/"+keyRing, &cloudkms.CryptoKey{
+			Purpose: "ASYMMETRIC_DECRYPT",
+			VersionTemplate: &cloudkms.CryptoKeyVersionTemplate{
+				Algorithm: "RSA_DECRYPT_OAEP_2048_SHA256",
+			},
+		}).CryptoKeyId("rsa-decrypt").Do()
+	kmsClient.Projects.Locations.KeyRings.CryptoKeys.Create(
+		parent+"/keyRings/"+keyRing, &cloudkms.CryptoKey{
+			Purpose: "ASYMMETRIC_SIGN",
+			VersionTemplate: &cloudkms.CryptoKeyVersionTemplate{
+				Algorithm: "RSA_SIGN_PSS_2048_SHA256",
+			},
+		}).CryptoKeyId("rsa-sign").Do()
+	kmsClient.Projects.Locations.KeyRings.CryptoKeys.Create(
```

Comment:
No need for the line wraps, Go has no line length limit.

Reply:
IMO the wrapped version is way more readable.